### PR TITLE
feat: limit maximum parcels per transaction

### LINKED ISSUE #385: Set a limit of N parcels to add to an Estate (due to gas limits)

### PR DESCRIPTION
**Pull request by cazala**
Closes #385 

This shows a warning if a single transaction reaches the limit of parcels, which is 12, ie: adding 12 parcels or removing 12 parcels. This doesn't mean 12 "modifications", because you can still add 10 parcels and remove 10 parcels in the same edition (since those are done in two different transactions).

This is an screenshot of how it looks like:

 
screen shot 2018-08-29 at 2 45 57 pm

This limit also applies when the user tries to dissolve an estate that is too big:

screen shot 2018-08-29 at 2 43 48 pm

### CODE REVIEW

File: webapp/src/components/EstateDetailPage/EstateSelect/EstateSelect.js [hunk 1]
```diff
@@ -101,13 +114,34 @@ export default class EstateSelect extends React.PureComponent {
...
     const parcelsToAdd = getParcelsNotIncluded(newParcels, pristineParcels)
+    return parcelsToAdd
+  }
+
+  getParcelsToRemove() {
+    const { estate, estatePristine } = this.props
+    const newParcels = estate.data.parcels
+    const pristineParcels = estatePristine.data.parcels
     const parcelsToRemove = getParcelsNotIncluded(pristineParcels, newParcels)
+    return parcelsToRemove
```
Comment (NicoSantangelo) [inline webapp/src/components/EstateDetailPage/EstateSelect/EstateSelect.js]: Maybe just return here?

`return getParcelsNotIncluded(pristineParcels, newParcels)`

File: webapp/src/components/EstateDetailPage/EstateSelect/EstateSelect.js [hunk 2]
```diff
@@ -101,13 +114,34 @@ export default class EstateSelect extends React.PureComponent {
...
   }
 
-  renderTxLabel = () => {
+  getParcelsToAdd() {
     const { estate, estatePristine } = this.props
     const newParcels = estate.data.parcels
     const pristineParcels = estatePristine.data.parcels
-
     const parcelsToAdd = getParcelsNotIncluded(newParcels, pristineParcels)
+    return parcelsToAdd
```
Comment (NicoSantangelo) [inline webapp/src/components/EstateDetailPage/EstateSelect/EstateSelect.js]: Maybe just return here

`return getParcelsNotIncluded(newParcels, pristineParcels)`

**Comment by NicoSantangelo** [comment]
@cazala Question, wouldn't this be a problem when using the `Delete` action for an Estate of >12 parcels?

**Comment by cazala** [comment]
@NicoSantangelo that makes sense, I tried to test it and I found out that it takes less gas to remove parcels than to add them to an estate, i dunno why. Creating an estate of 24 parcels consumes around 4.5M gas (and fails with an `Out of gas` exception) while deleting an estate of 24 parcels requires 4.2M gas and the tx succeeds. But in theory if I try to delete a bigger estate (say, 30 parcels) that should also throw an OOG, I'll try to get more ropsten LAND and verify it.

But in short, yeah, we should apply a limit for the `Delete` action too.

**Comment by nachomazzara** [comment]
I think we should change the message explaining that it is not recommended instead of impossible using “can’t”.

Like:

You reach the limit of 12 parcels. We use this limit because the transaction can run out of gas based on the limit the evm has nowadays.

File: webapp/src/components/EstateDetailPage/EstateSelect/EstateSelect.js [hunk 3]
```diff
@@ -13,6 +20,8 @@ import { hasNeighbour, areConnected, isEstate } from 'shared/estate'
 import { getParcelsNotIncluded } from 'shared/utils'
 import './EstateSelect.css'
 
+const MAX_PARCELS_PER_TX = 12
```
Comment (nachomazzara) [inline webapp/src/components/EstateDetailPage/EstateSelect/EstateSelect.js]: What do you think of moving this constant to share/estates?

**Pull request merged**